docsk8s/troubleshooting/ls-k8s-common-problems.asciidoc

```diff
@@ -1,25 +1,59 @@
 [[ls-k8s-common-problems]]
 === Common problems
 
-Following are some suggested resolutions to problems that you may encounter when running Logstash in a Kubernetes environment.
+Following are some suggested resolutions to problems that you may encounter when running {ls} in a Kubernetes environment.
 
-* <<problem-01>>
-* <<problem-02>>
+* <<problem-keep-restart>>
+* <<problem-oom>>
 
 [float]
-[[problem-01]]
+[[problem-keep-restart]]
 === Logstash keeps restarting
 When you check the running Kubernetes pods status, {ls} shows continual restarts. 
 
-This can be caused by...
+[source,bash]
+--
+NAMESPACE     NAME                                  READY   STATUS    RESTARTS      AGE
+default       logstash-f7768c66d-grzbj              0/1     Running   3 (55s ago)   6m32s
+--
 
-To resolve the problem...
+This can be caused by a few issues:
 
 [float]
-[[problem-02]]
-=== Logstash stops with an OOM errors
-The {ls} pod continues to stop running due to an out of memory (OOM) error.
+[[problem-nometric]]
+==== Metrics API not accessible to `readinessProbe`
 
-This can be caused by...
+If the `readinessProbe` is unable to access the health check endpoint, the {ls} process will be continuously stopped and restarted. To fix it, set the following in `logstash.yml` in `ConfigMap`.
 
-To resolve the problem...
+[source,bash]
+--
+api.http.host: 0.0.0.0
+--
+
+[float]
+[[problem-delay]]
+==== {ls} startup process takes longer than `initialDelaySeconds`
+
+It may be necessary to review the time constraints of `readinessProbe` and `livenessProbe` to ensure that {ls} has enough time to start up and expose the health check endpoint for the `readiness` and `liveness` probes to access.
+
+[float]
+[[problem-insufficient]]
+==== Insufficient CPU or memory to start {ls}
+
+Review CPU and memory usage using `kubectl top pods` (note that this requires metrics server to be available for your Kubernetes implementation)
+
+* Set the values of `cpu` and `memory` in your `Deployment` or `StatefulSet` appropriately.
+* Ensure that the JVM memory settings are set appropriately, note that the default `Xmx` value is `1g`, and we recommend that heap size is set to no more than 50-75% of total memory.
+
+
+
+[float]
+[[problem-oom]]
+=== {ls} stops with OOM errors
+The status of {ls} shows `Ready` but the pod continues to stop running.
+
+This can be caused by insufficient memory. If the memory of {ls} use more than the declared resource, kubernetes kill the pod immediately and the log of {ls} does not show any shutdown related message.
```

Comment:
```suggestion
This situation can be caused by insufficient memory. If {ls} uses more memory than the declared resource, Kubernetes kills the pod immediately and the l{ls} log does not show any shutdown related message.
```

Reply:
Kubernetes should be capitalized in every mention.

Reply:
We try to use less violent language like "stopped" or "shut down" instead of "killed."  I see your "OOMKilled" message later in the docs, so I'm guessed that common language for K8s. Still, we should try to avoid when we can do so without confusing users.

Reply:
we can use "Kubernetes shutdowns the pod" instead. "OOMKilled" is a pod status. "kill" is a common language in OS, eg. kill -9 process_id, to send a kill signal to terminate the process immediately